pkg/kubectl/rolling_updater.go

```diff
@@ -208,9 +208,15 @@ func (r *RollingUpdater) Update(config *RollingUpdaterConfig) error {
 	if err != nil {
 		return err
 	}
-	// Further validation.
-	if maxUnavailable == 0 && maxSurge == 0 {
-		return fmt.Errorf("one of maxSurge or maxUnavailable must be specified")
+	// If the desired new scale is 0, then the max unavailable is necessarily
+	// the effective scale of the old RC regardless of the configuration
+	// (equivalent to 100% maxUnavailable).
+	if desired == 0 {
+		maxUnavailable = int(math.Max(float64(oldRc.Spec.Replicas), float64(oldRc.Status.Replicas)))
+	} else {
+		if maxUnavailable == 0 && maxSurge == 0 {
```

Comment:
fwiw, I was trying to find if the same problem exists with deployments and it does not because we do not check the computed values of maxSurge and maxUnavailable there, but check the original values instead
https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/extensions/validation/validation.go#L263.